FFM 9030 - Add Polling (#98)
harness/ff-javascript-client-sdk: README.md

  baseUrl?: string
  eventUrl?: string
  eventsSyncInterval?: number

>   pollingInterval?: number

Reviewer: I'd add a `pollingEnabled` as well. It would also be helpful to add a section explaining Polling and Streaming, detailing how they're used and configured.

Author: Done - now polling and streaming is opt in. As we discussed: "Streaming takes precedence, but falls back to polling unless disabled"

>   pollingInterval?: number

Reviewer: ```suggestion
  pollingInterval?: number
  pollingEnabled?: boolean
```

})
```

## Streaming and Polling Mode

> By default, Harness Feature Flags SDK has streaming enabled and polling disabled. Both modes can be toggled according to your preference using the SDK's configurations.

Reviewer: Generally I'd use `configuration` instead of `configurations`. The configuration is a single entity which contains multiple configuration items.

Author: Updated

### Streaming Mode

> Streaming mode establishes a continuous connection between your application and Harness' servers. This allows for real-time updates on feature flags without requiring periodic checks. If an error occurs while streaming and `pollingEnabled` is set to `true`, the SDK will automatically fall back to polling mode until streaming can be reestablished. If `pollingEnabled` is `false`, streaming will attempt to reconnect without falling back to polling.

Reviewer: > continuous connection between your application and Harness' servers

This might be a little too specific for those who use the Proxy. Maybe instead of `Harness' servers` something like `the Feature Flags service`?

Author: Updated

### Polling Mode

> In polling mode, the SDK will periodically check with the Harness servers to retrieve updates for feature flags. The frequency of these checks can be adjusted using the SDK's configurations.

Reviewer: ```suggestion
In polling mode, the SDK will periodically check with the Feature Flags service to retrieve updates for feature flags. The frequency of these checks can be adjusted using the SDK's configurations.
```

### No Streaming or Polling
If both streaming and polling modes are disabled (`streamEnabled: false` and `pollingEnabled: false`), 
the SDK will not automatically fetch feature flag updates after the initial fetch. This means that after the initial load, any changes made to the feature flags on the Harness server will not be reflected in the application until the SDK is re-initialized or one of the modes is re-enabled.

This configuration might be useful in specific scenarios where you want to ensure a consistent set of feature flags 
for a session or when the application operates in an environment where regular updates are not necessary. However, it's essential to be aware that this configuration can lead to outdated flag evaluations if the flags change on the server.

> To use the SDK with both modes disabled:

Reviewer: I think this sentence might be deletable

Author: Whoops - deleted

### Configurations
While both streaming and polling can be enabled, if both are active, streaming takes precedence. 

> If streaming encounters an issue and `pollingEnabled` is set to `true`, the SDK will switch to polling mode as a backup until streaming is recovered. If `pollingEnabled` is set to `false`, the SDK will continue attempting to reconnect via streaming.

Reviewer: I think you covered this up in the Streaming section

Author: Oops - deleted

To configure the modes:

```typescript

const options = {
  streamEnabled: true, // Enable or disable streaming - default is enabled

>   pollingEnabled: false, // Enable or disable polling - default is disabled. Enable to fallback to polling if streaming encounters an issue.

Author: ```suggestion
  pollingEnabled: false, // Enable or disable polling - default is disabled. Enable to fallback to polling if streaming is enabled encounters an issue.
```

}

> const client = initialize(
>         'YOUR_SDK_KEY',
>         {
>           identifier: 'Harness1',
>           attributes: {
>             lastUpdated: Date(),
>             host: location.href
>           }
>         },
>         options,
> 
> )

Reviewer: The indentation is way off here and has a dangling comma
```suggestion
const client = initialize(
  'YOUR_SDK_KEY',
  {
    identifier: 'Harness1',
    attributes: {
      lastUpdated: Date(),
      host: location.href
    }
  },
  options
)
```
